tests/assets/voc_dataset/voc_dataset2/ImageSets/Action/trainval.txt

```diff
@@ -1,3 +1,4 @@
+@
```

Comment:
What is the purpose in calling this file this way?

Reply:
It's so that it would come before A in the sorted list. I should probably add a comment about this.

Reply:
Can we have a dedicated test for this?

Reply:
I expanded the comment inside `@.xml`.

Reply:
It's quite easy to overlook breaking changes, which will lead to regression in this question. There is no specific test for sorting, and if we remove sorting everything can work well depending on platform logic for returned values from file list fetching. VOC auto detection is a selling feature for some customers, so I think it's better to be protected with a bunch of tests around, stating the expected behavior and making sure we covered corner cases. I'll leave it up to you to decide on this, maybe it will be better to add such tests when we cover this case by some kind of logic described in my proposal.

Reply:
> There is no specific test for sorting, and if we remove sorting everything can work well depending on platform logic for returned values from file list fetching.

We unfortunately cannot decisively verify that sorting works, since we can't be sure that the filesystem on the testing machine doesn't give us the directory entries in sorted order to begin with. `ApplyFormatDetectorTest.test_require_files_success` is the best we can do WRT verifying this.

However, even if the sorting is accidentally broken and we don't catch it, that won't cause a regression with regards to #658. The sorting only exists to make the detector more deterministic (and thus make the tests more deterministic as well). For a regression to happen, the LabelMe detector would _also_ have to be broken (but in such a way that both `VocImportTest.test_can_detect_voc` and `LabelMeImporterTest.test_can_detect` still pass). So I think we're protected against regression as well as we can be.

Also, this argument doesn't really seem to support your original suggestion - a dedicated test would just have the same problem.

> VOC auto detection is a selling feature for some customers, so I think it's better to be protected with a bunch of tests around, stating the expected behavior and making sure we covered corner cases.

I don't necessarily disagree, but this isn't a general "improve VOC detection testing" PR - I'm just trying to fix a specific scenario (#658). And since I can piggyback on an existing test dataset, I don't really see why I shouldn't do that. I think the new test annotation file is reasonably documented, so there should be no confusion as to what it's for.

> The example is that we have only XML files with image metadata.

That would be a different scenario, so I think it's off-topic.